### PR TITLE
Testing branch

### PR DESCRIPTION
**Pull request by StephanieFriend**
- In setup changed the standard initialize method into a manual method
- Also in setup altered the welcome message methods and removed the quit method. 
- The setup tests will have to be fully reviewed and possibly altered 
- Created the turn class to produce the board with a proper loop that will allow for replay
- Refactored the turn class to remove recursion and allow for a better game play

### CODE REVIEW

**Comment by StephanieFriend** [comment]
@cmpprg  please review :)

File: lib/turn.rb [hunk 1]
```diff
@@ -53,16 +70,61 @@ def validate_coordinate_exists?(coordinate)
   end
 
   def validate_cell_fired_upon?(coordinate)
-    if !@setup.computer_board.not_fired_upon.include?(coordinate)
+    if @setup.computer_board.cells[coordinate].fired_upon?
```
Comment (cmpprg) [inline lib/turn.rb:73]: implement not_fired_upon method from board class please.

File: lib/turn.rb [hunk 2]
```diff
@@ -53,16 +70,61 @@ def validate_coordinate_exists?(coordinate)
...
     end
   end
 
-  def comp_shot_implementation
-    comp_shot = @setup.player_board.coordinate_not_fired_upon
+  def computer_shot_implementation
+    comp_shot = ""
+    loop do
+      comp_shot = @setup.player_board.coordinate_not_fired_upon
+      break if !@setup.player_board.cells[comp_shot].fired_upon?
```
Comment (cmpprg) [inline lib/turn.rb:85]: implement coordinate_not_fired_upon? method from board class.

File: lib/turn.rb [hunk 3]
```diff
@@ -53,16 +70,61 @@ def validate_coordinate_exists?(coordinate)
...
+
+  def print_cell_results(render)
+    return "was a miss" if render == "M"
+    return "was a hit" if render == "H"
+    "sunk a ship"
+  end
+
+  def shot_results(player, computer)
+    player_render = print_cell_results(@setup.computer_board.cells[player].render)
+    computer_render = print_cell_results(@setup.player_board.cells[computer].render)
```
Comment (cmpprg) [inline lib/turn.rb:99]: if time create new method in board to refactor.

**Review by cmpprg — CHANGES_REQUESTED** [review]
please review comments and make changes to turn class. Everything looks really good.

File: lib/turn.rb (hunk 1 above)
Comment (StephanieFriend) [reply lib/turn.rb:73]: there is no boolean method in board class for fired_upon?
maybe we can review this as a refactor after we merge?

File: lib/turn.rb (hunk 2 above)
Comment (StephanieFriend) [reply lib/turn.rb:85]: there is no boolean method in board class for fired_upon?
maybe we can review this as a refactor after we merge?

**Comment by StephanieFriend** [comment]
@cmpprg  I made comments. Please review

File: lib/turn.rb (hunk 1 above)
Comment (cmpprg) [reply lib/turn.rb:73]: sounds great

File: lib/turn.rb (hunk 2 above)
Comment (cmpprg) [reply lib/turn.rb:85]: I will create this method tonight, we will merge for now and refactor later.

**Comment by cmpprg** [comment]
Almost there.........

**Pull request merged**